### DISCUSSION TITLE
Large difference between kpi with/without optimization

### DISCUSSION THREAD

**Discussion by jhpkr** [google/lightweight_mmm#40] [Q&A]
Hello, 

I have been following the demos and examples of lightweight and have been applying lightweight to my own dataset. The value of this variable is significantly lower than what I had expected. Given the dataset, I had expected to be around the 2000 range.

<img width="691" alt="Screen Shot 2022-07-12 at 2 00 20 PM" src="https://user-images.githubusercontent.com/106620040/178561780-b25abff8-a986-4340-b5bc-fa01da8ff8f0.png">

To increase this value, I have tried to rescale the values as well as switching the budgets but have not been able to find any success. Could I get some more insight into which variables impact the kpi_without_optimization variable?

**Comment by pabloduque0** [comment]
Hello @jhpkr !

Is the problem that the value is low or that the difference from with to without optim is too large?

A couple questions to try to understand what is going on?
- Did you use extra features? If so, what kind and how did you pass it to the optimization?
- Did you scale the data? If so, how?

We have not yet build the utility to get insights for any given predictions but we will probably add that at some point. In the mean time you can always "bake" it yourself by using the trace of your model and the data you passed.

What you can also do that can be quite informational is to pass the output of the optimization to the response curves as `optimal_allocation_per_timeunit` (you might need to divide it first by `n_time_periods` depending for how long you are optimizing for). [See documentation here](https://lightweight-mmm.readthedocs.io/en/latest/api.html#lightweight_mmm.plot.plot_response_curves).

**Reply by jhpkr** [reply]
Hello @pabloduque0 ,

The problem is that the **kpi_without_optim** variable is too low. For this model, I have not used any extra features (i.e. promotions, competitor data, etc.). I scaled the data by the mean of non-zero values given the number of 0s in some media variable. 

I will definitely look into passing the output of the optimization to the response curves and tracing my data.

**Reply by pabloduque0** [reply]
The `kpi_without_optim` is calculated by running a prediction based on an average of the training data. If your data distribution is a bit skewed and the mean is not what you are looking for you can always create your "assumed media" and get a prediction based on that. Then you can compare that to the `optimized` output.

**Reply by jhpkr** [reply]
Hi @pabloduque0,

Thank you for the responses. 

You mentioned that the kpi_without_optim is calculated by running a prediction based on an average of the training data. However, when looking at the average installations from the training data, the mean is around 2400. In addition to this, model seems to show that the predicted values are very close the actual values.

Training Model:
<img width="500" alt="Screen Shot 2022-07-15 at 2 05 36 PM" src="https://user-images.githubusercontent.com/106620040/179285126-1255de18-3ada-4006-91b2-b16bdbd1f133.png">

Predicted Model:
<img width="408" alt="Screen Shot 2022-07-15 at 2 05 18 PM" src="https://user-images.githubusercontent.com/106620040/179285143-eaf43122-87e4-4551-9110-589e1b14a76f.png">


 I am having trouble understanding how kpi_without_optim could produce this value given these two factors above. Could I get some assistance on how to proceed?

**Reply by pabloduque0** [reply]
The negative in the results you showed is only to indicate the optimisation since we minimize in it. I think you know that but just in case the confusion is coming from that part.

What would be your expected outputs for some of these values?

**Reply by jhpkr** [reply]
Hi @pabloduque0 ,

Given that kpi_without_optim is calculated by running a prediction based on an average of the training data, I had expected the value to be at least around 2000... but the value that I currently have (1068) is very far from that. Taking to consideration on our earlier statements, would it be fair to compare the optimized kpi variable (3200) with the mean of the dataset (2400)?

**Reply by pabloduque0** [reply]
Sorry for the late reply. 

Well by comparing to the mean of the dataset you are not taking into account a few seasonality factors and it is best to compare it to another prediction with whatever values for media you consider as average.
